gateway/src/resty/http/chunked.lua

```diff
@@ -0,0 +1,161 @@
+local co_wrap_iter = require("resty.coroutines").co_wrap_iter
```

Comment:
Why are we reimplementing the library here? Surely we should leverage the functions available in [lua-resty-http](https://github.com/ledgetech/lua-resty-http/blob/master/lib/resty/http.lua#L971-L1001)?

Reply:
@kevprice83 done with the unittest.

because the lua-resty-http is writtent as a decoder and only send the data body
https://github.com/3scale/APIcast/blob/master/gateway/src/apicast/http_proxy.lua#L168

Reply:
Sorry, I do not get that. https://github.com/3scale/APIcast/blob/master/gateway/src/apicast/http_proxy.lua#L168 is only about rewriting request host header.

Reply:
Sorry I put wrong link there, I mean this one https://github.com/ledgetech/lua-resty-http/blob/master/lib/resty/http.lua#L480

Reply:
Ok, I see. I found this PR [Add support for chunked body to _send_body #285](https://github.com/ledgetech/lua-resty-http/pull/285) as alternative. I also like the idea of the our own wrapper while not in lua-resty-http

Reply:
Added tests for:
* http proxy env vars (over TLS)
* http_proxy policy (over TLS)
* camel_proxy policy (over TLS)